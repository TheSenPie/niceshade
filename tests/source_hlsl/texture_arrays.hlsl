//T:texture_arrays cs:CSMain


[[vk::binding(0,0)]] Texture2D<float> inTexArr[4];
[[vk::binding(1,0)]] sampler samp;
[[vk::binding(2,0)]] Texture2D<float> inTex;
[[vk::binding(3,0)]] RWTexture2D<float> output;

[numthreads(16,16,4)]
void CSMain(uint3 dtid : SV_DispatchThreadID) {
  float x = inTexArr[dtid.z].SampleLevel(samp, dtid.xy / 2.0f, 0, 0).x;
  float y = inTex.SampleLevel(samp, dtid.xy / 2.0f, 0, 0).x;
  output[dtid.xy] = x * y;
}

