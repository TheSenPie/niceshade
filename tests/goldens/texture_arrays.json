{
"header": {
  "magic_number": 3735928559,
  "header_size": 40,
  "version_maj": 0,
  "version_min": 1,
  "entrypoints_offset": 40,
  "pipeline_layout_offset": 64,
  "image_to_cis_map_offset": 120,
  "sampler_to_cis_map_offset": 156,
  "user_metadata_offset": 180
},
"entrypoints": { 
  "vertex": "(null)",
  "fragment": "(null)",
  "compute": "CSMain"
},
"pipeline_layout": {
  "descriptor_sets": [
    {
      "set": 0,
      "descriptors": [
        {
          "binding": 0,
          "type": "IMAGE",
          "stage_vis": 4
        },
        {
          "binding": 1,
          "type": "SAMPLER",
          "stage_vis": 4
        },
        {
          "binding": 2,
          "type": "IMAGE",
          "stage_vis": 4
        },
        {
          "binding": 3,
          "type": "LOADSTORE_IMAGE",
          "stage_vis": 4
        }
      ]
    }
  ]
},
"image_to_cis_map": {
  "entries": [
    {
      "entry": 0,
      "separate_set_id": 0,
      "separate_binding_id": 0,
      "combined_ids": [0]
    },
    {
      "entry": 1,
      "separate_set_id": 0,
      "separate_binding_id": 2,
      "combined_ids": [1]
    }
  ]
},
"sampler_to_cis_map": {
  "entries": [
    {
      "entry": 0,
      "separate_set_id": 0,
      "separate_binding_id": 1,
      "combined_ids": [0, 1]
    }
  ]
},
"threadgroup_size": [16, 16, 4],
"user_metadata": {
}
}
